### DISCUSSION TITLE
大文件上传失败

### DISCUSSION THREAD

**Discussion by PastWind-Zhou** [AlistGo/alist#2130] [Q&A]
![Screenshot_2022-10-26-18-48-14-381_com microsoft rdc android-edit](https://user-images.githubusercontent.com/34813628/198025370-f98b8254-adfe-4527-a0e8-2c2267991cdc.jpg)
上传大文件，在最后报错提示Network Error，挂载的是one drive和阿里云盘，本地搭建和云主机搭建都有这个问题

**Comment by PastWind-Zhou** [comment]
用stream上传大文件提示The requested file could not be read, typically due to permission problems that have occurred after a reference to a file was acquired.
用form上传大文件传到最后就报NetworkError

**Comment by GengGode** [comment]
同，本地和天翼网盘都会出现上传失败
![image](https://user-images.githubusercontent.com/47711102/198087862-94ac0cd2-c953-4c05-a9e5-512da5d0c163.png)
只能上传成功小于1Mb的文件，很奇怪，并不分网盘类型，似乎都有

**Reply by PastWind-Zhou** [reply]
我这边试过600M的没问题，但是1.8G的就报错

**Reply by GengGode** [reply]
非常奇怪，是什么参数控制的吗

**Comment by sky121666** [comment]
![image](https://user-images.githubusercontent.com/82988354/198309597-7ba7ee4d-a1e4-42d8-a0fc-f658cd570698.png)用stream上传必报错


### **Form 上传方式  ：**
![image](https://user-images.githubusercontent.com/82988354/198310610-273bbb82-7114-4691-8f39-a746dce173ec.png)
3.4GB windows7 iso 本地上传 无问题，上传本地，上传阿里都没有问题。
下图是
4.6GB  Centos7 
5.2GB  win Server 2022
大于4GB 无法上传，不知道在哪调整上传限制
![image](https://user-images.githubusercontent.com/82988354/198311673-9c6cd1a5-0c33-4c09-b5ba-f1f125b420c8.png)

aria2 模式下载后自己上传到阿里云盘无问题，从本地网盘 复制到阿里云盘无问题
![image](https://user-images.githubusercontent.com/82988354/198312701-033f851d-f576-404f-96ba-62e220581805.png)


遇到的问题就是只能Form上传，以及大约4GB 上传的限制

**Comment by sky121666** [comment]
问题解决。增加tmp 大小即可解决4GB大小限制

**Reply by GengGode** [reply]
在哪里，求指路，感谢

**Reply by sky121666** [reply]
> 在哪里，求指路，感谢
解不了你的惑了，遇到的问题不同。我这个是tmp 相当于ram的一半是4g，so。我这个加个内存条或者调个比例就ok了。你的网络错误搞不了，我这个是群晖内装的

**Reply by sky121666** [reply]
> 在哪里，求指路，感谢
![捕获](https://user-images.githubusercontent.com/82988354/198368121-04195377-4c43-4546-a2aa-769e665e101b.PNG)
![捕获1](https://user-images.githubusercontent.com/82988354/198368122-d85c66ca-d96c-4afb-a941-09e07cdff3ed.PNG)
我这个就是把tmp 改了个比例，然后这个大约6g的iso包就可以上传了，这个搜了一下tmp是根据内存来的，会不会溢出不知道，这方面不太懂

**Reply by GengGode** [reply]
悲，确实不一样，寄寄子了

**Comment by haodong1668** [comment]
123盘,From上传方式，到最后的时候报错500,这是什么问题，有大佬会解决吗？
![image](https://user-images.githubusercontent.com/103712417/199384636-8900213d-c07a-4885-a5fe-c08c56eb93a3.png)
